src/main/java/uk/gov/hmcts/reform/coh/service/QuestionService.java

```diff
@@ -111,4 +125,52 @@ public Question updateQuestionForced(Question question){
     public void deleteQuestion(Question question) {
         questionRepository.delete(question);
     }
+
+    @Transactional
+    public void requestDeadlineExtension(OnlineHearing onlineHearing) {
+        List<Question> questions = findAllQuestionsByOnlineHearing(onlineHearing)
+            .orElseThrow(() -> new RuntimeException("Could not retrieve questions"));
+
+        if (questions.isEmpty()) {
+            throw new RuntimeException("There are no questions to be answered");
+        }
+
+        Instant now = Instant.now();
+
+        questions.stream()
+            .filter(question -> now.isBefore(question.getDeadlineExpiryDate().toInstant()))
+            .forEach(question -> {
+                if (shouldGrantExtensionTo(question)) {
+                    grantDeadlineExtension(question);
+                } else if (shouldDenyExtensionTo(question)) {
+                    denyDeadlineExtension(question);
+                }
+            });
+    }
+
+    private boolean shouldGrantExtensionTo(Question question) {
+        return issued.equals(question.getQuestionState());
+    }
+
+    private boolean shouldDenyExtensionTo(Question question) {
+        return extensionGranted.equals(question.getQuestionState());
+    }
+
+    private void grantDeadlineExtension(Question question) {
+        question.setDeadlineExpiryDate(newDeadline(question));
+        question.setQuestionState(extensionGranted);
+        question.updateQuestionStateHistory(extensionGranted);
+        questionRepository.save(question);
+    }
+
+    private Date newDeadline(Question question) {
+        Temporal temporal = extension.addTo(question.getDeadlineExpiryDate().toInstant());
+        return Date.from(Instant.from(temporal));
+    }
```

Comment:
The new deadline should be 7 days in the future. For now, make it a configurable value in the application.yaml files